### PR TITLE
Prepare for 4.0.0 development

### PR DESCRIPTION
**Pull request by erdlet**
Since the new API adds a new interface method to MvcContext, we already need to implement it in this PR.

see: #367

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -44,8 +44,8 @@
         <maven.compiler.release>11</maven.compiler.release>
 
         <!-- DEPENDENCY PROPERTIES -->
-        <spec.version>2.1.0</spec.version>
-        <tck.version>2.1.0</tck.version>
+        <spec.version>3.0.0.M1</spec.version>
+        <tck.version>3.0.0.M1</tck.version>
 
         <jersey.version>3.1.0-M8</jersey.version>
```
Comment (ivargrimstad) [inline pom.xml:50]: Update jersey version to 3.1.1 ?

**Review by ivargrimstad — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<!--
+    <!--
```
Comment (mthmulders) [inline pom.xml:1]: Nit: probably not necessary?

File: pom.xml (hunk 1 above)
Comment (erdlet) [reply pom.xml:50]: Yes good catch. Just focused on the spec yesterday. I'll change it

File: pom.xml (hunk 2 above)
Comment (erdlet) [reply pom.xml:1]: Thanks for the review, I'll fix it later👍🏻

**Review by chkal — COMMENTED** [review]
Other than the two aspects already mentioned, everything looks good to me. Thanks a lot for the effort.

**Review by ivargrimstad — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (erdlet) [reply pom.xml]: Done

**Review by ivargrimstad — APPROVED** [review]

**Review by chkal — APPROVED** [review]

**Comment by erdlet** [comment]
Removed Krazo and MVC from integration tests to avoid build errors

**Pull request merged**